### DISCUSSION TITLE
Changing the shopify private app/store that hydrates Statamic?

### DISCUSSION THREAD

**Discussion by austriker27** [statamic-rad-pack/shopify#80] [Help] [answered]
Is it okay and possible to change the shopify store that's hydrating Statamic? I ask because I setup my repo and linked it successfully to a shopify test account. However when I change to my new production store I get `Something went wrong. Please try again.` when I try to import the products. 

Here's what I did:

1. Removed all shopify related entries in statamic (products, variations, blueprints, taxonomies, assets, php files, JS files)
2. Uninstall shopify statamic addon (just for good measure)
2. Created private app in new shopify store via docs
3. Updated ENV variables with new data
4. Added back JS file imports into `site.js` 
5. Restart yarn watch, clean cache just for good measure

I even created a new private app in the shopify store and tried its credentials. 

I wonder if its something wrong with my shopify store. The only difference is the prior shopify shop was a dev shop I created that never was activated - Ive been using my current real shop for a few weeks via the POS and thus its been activated.

EDIT:

More updates, I think I've narrowed it down to the `product_type` taxonomy: 

The top of my stack trace for today says: 
```
[2021-08-14 00:13:25] local.DEBUG: Cannot render an array variable as a string: {{ product_type }}  
[2021-08-14 00:14:37] local.DEBUG: Cannot loop over non-loopable variable: {{ product_type }}  
[2021-08-14 00:15:07] local.DEBUG: Cannot loop over non-loopable variable: {{ product_type }}  
[2021-08-14 00:15:13] local.DEBUG: Cannot render an array variable as a string: {{ product_type }}  
[2021-08-14 00:18:05] local.DEBUG: Cannot loop over non-loopable variable: {{ product_type }}  
[2021-08-14 00:25:15] local.ERROR: Call to a member function field() on null {"userId":"6bbb8ea9-e8f0-4906-9f24-af95d70f961c","exception":"[object] 
```

Which is confusing to me, because i'm only ever looping over `product_type` in my template files. And the id `6bbb8ea9-e8f0-4906-9f24-af95d70f961c` is my personal user, which looks like fine to me.

**Comment by jackabox** [comment]
You can swap the env at any point and it’ll just import from the new shop next time. It’s advisable to delete all content before swapping but you shouldn’t have to change anything else.

Chances are your app doesn’t have the right privileges

Many Thanks,
Jack Whiting

https://jackwhiting.co.uk
Freelance PHP & JS Developer

**Reply by austriker27** [reply]
Okay, that's what I thought but I tried everything I could think of to fix this import bug situation I'm in.... So that's good to have confirmed that changing shops is as easy as changing ENVs. Ive been going crazy trying to fix this 😕 🤪

_(Sidenote: maybe a new FR to have the addon return 'credentials not correct' or something)_


I've triple checked the new private app, do you mind checking this to see if I missed anything? Here's my permissions, am I missing anything? 
![image](https://user-images.githubusercontent.com/14855925/129481877-6a71b7bc-9853-41b2-9ea0-2ddc2350857c.png)
![image](https://user-images.githubusercontent.com/14855925/129481897-af00da27-3f18-4aea-a909-fd2627ba47c6.png)

**Comment by austriker27** [comment, marked as answer]
Okay Ive figured it out and its importing now!! For anyone else who comes across this issue - make sure your container is set to be the same thing everywhere that the shopify addon touches (blueprints, etc) because mine was different in the ENV and the `product_type` blueprint and thus I had issues.
